tests/test_qaoa.py

```diff
@@ -193,12 +240,52 @@ def decompose_hamiltonian(hamiltonian):
 class TestCostHamiltonians:
     """Tests that the cost Hamiltonians are being generated correctly"""
 
+    """Tests the cost Hamiltonian components"""
+
+    def test_bit_driver_error(self):
+        """Tests that the bit driver Hamiltonian throws the correct error"""
+
+        with pytest.raises(ValueError, match=r"'state' argument must be either 0 or 1"):
+            qaoa.bit_driver(range(3), 2)
+
+    def test_bit_driver_output(self):
+        """Tests that the bit driver Hamiltonian has the correct output"""
+
+        H = qaoa.bit_driver(range(3), 1)
+        hamiltonian = qml.Hamiltonian([1, 1, 1], [qml.PauliZ(0), qml.PauliZ(1), qml.PauliZ(2)])
+
+        assert decompose_hamiltonian(H) == decompose_hamiltonian(hamiltonian)
+
+    def test_edge_driver_errors(self):
+        """Tests that the edge driver Hamiltonian throws the correct errors"""
+
+        with pytest.raises(ValueError, match=r"Encountered invalid entry in 'reward', expected 2-bit bitstrings."):
+            qaoa.edge_driver(Graph([(0, 1), (1, 2)]), ['10', '11', 21, 'g'])
+
+        with pytest.raises(
+                ValueError,
+                match=r"'reward' cannot contain either '10' or '01', must contain neither, or both."
+        ):
+            qaoa.edge_driver(Graph([(0, 1), (1, 2)]), ['11', '00', '01'])
+
+        with pytest.raises(ValueError, match=r"Input graph must be a nx.Graph"):
+            qaoa.edge_driver([(0, 1), (1, 2)], ['00', '11'])
+
+    @pytest.mark.parametrize(("graph", "reward", "hamiltonian"), EDGE_DRIVER)
+    def test_edge_driver_output(self, graph, reward, hamiltonian):
+        """Tests that the edge driver Hamiltonian throws the correct errors"""
+
+        H = qaoa.edge_driver(graph, reward)
+        assert decompose_hamiltonian(H) == decompose_hamiltonian(hamiltonian)
+
+    """Tests the cost Hamiltonians"""
+
```

Comment:
Are the tests for max clique, min vertex cover, and max indep set in other PRs?

Reply:
No, this is still a WIP PR, so I haven't finished writing them quite yet